firmware/pattern.c

```diff
@@ -16,6 +16,7 @@ unsigned int pattern_framebuffer_base(void) {
 	return FRAMEBUFFER_BASE_PATTERN;
 }
 
+#ifdef MAIN_RAM_BASE
```

Comment:
I'm not sure why this define is here? Being `static const` should mean it doesn't end up in the SRAM?